python/ray/tests/test_client_terminate.py

```diff
@@ -88,6 +89,39 @@ def wait_for(t):
         ray.get(obj1)
 
 
+def test_kill_cancel_metadata(ray_start_regular):
+    """
+    Verify that client worker's terminate_actor and terminate_task methods
+    pass worker's metadata attribute server to the grpc stub's Terminate
+    method.
+    """
+    with ray_start_client_server(metadata=[("foo", "bar")]) as ray:
+
+        @ray.remote
+        class A:
+            pass
+
+        @ray.remote
+        def f():
+            time.sleep(1000)
+
+        actor = A.remote()
+        task_ref = f.remote()
+
+        def mock_terminate(term, metadata):
+            raise Exception(metadata[0][0])
```

Comment:
```suggestion
            raise MetaDataIsCorrectlyPassedException(metadata[0][0])
```

Reply:
Raising an exception here is properly not the right idiom at all.
I think `unittest` provides a framework for side effects to do this. Will take a brief read.

Reply:
eh, too lazy, will define a more descriptive Exception class, as you've suggested